File: server/shoppe/jv-service/src/main/java/com/lcaohoanq/jvservice/domain/chat/ChatMessage.java
package com.lcaohoanq.jvservice.domain.chat;

import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatMessage {
    private Long id;
    private Long chatRoomId;
    private Long senderId;
    private String content;
    private String attachmentUrl;
    private LocalDateTime timestamp;
    private boolean isRead;
    
} 

File: chat-service/src/main/kotlin/com/lcaohoanq/chatservice/domain/chat/ChatMessage.kt
package com.lcaohoanq.chatservice.domain.chat

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "chat_message")
class ChatMessage {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private var id: Long = 0

    @Column(name = "chat_id")
    private var chatId: String = ""

    @Column(name = "sender_id")
    private var senderId: String = ""

    @Column(name = "recipient_id")
    private var recipientId: String = ""

    private var content: String = ""

    private var timestamp: Date = Date()

    fun getId(): Long {
        return id
    }

    fun getSenderId(): String {
        return senderId
    }

    fun setChatId(chatId: String) {
        this.chatId = chatId
    }

    fun getRecipientId(): String {
        return recipientId
    }

    fun getContent(): String {
        return content
    }

    fun getTimestamp(): Date {
        return timestamp
    }
}
